sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelHealthChecker.java

```diff
@@ -206,6 +206,122 @@ public Future<Boolean> isHealthy(final Channel channel) {
+                    "lastChannelWriteNanoTime: {}, writeDelayInNanos: {}, writeDelayLimitInNanos: {}, " +
+                    "rntbdContext: {}, pendingRequestCount: {}}",
+                channel, timestamps.lastChannelWriteAttemptNanoTime(), timestamps.lastChannelWriteNanoTime(),
+                writeDelayInNanos, this.writeDelayLimitInNanos, rntbdContext, pendingRequestCount);
+
+            String msg = MessageFormat.format(
+                "{0} health check failed due to nonresponding write: (lastChannelWriteAttemptNanoTime: {1}, " +
+                    "lastChannelWriteNanoTime: {2}, writeDelayInNanos: {3}, writeDelayLimitInNanos: {4}, " +
+                    "rntbdContext: {5}, pendingRequestCount: {6})",
+                channel, timestamps.lastChannelWriteAttemptNanoTime(), timestamps.lastChannelWriteNanoTime(),
+                writeDelayInNanos, this.writeDelayLimitInNanos, rntbdContext, pendingRequestCount
+            );
+
+            return promise.setSuccess(msg);
+        }
+
+        // Black hole detection, part 2:
+        // Treat the connection as unhealthy if the gap between the last successful write and the last successful read
+        // grew beyond acceptable limits, unless a write succeeded recently. This is a sign of a nonresponding read.
+
+        final long readDelay = timestamps.lastChannelWriteNanoTime() - timestamps.lastChannelReadNanoTime();
+        final long readHangDuration = currentTime - timestamps.lastChannelWriteNanoTime();
+
+        if (readDelay > this.readDelayLimitInNanos && readHangDuration > readHangGracePeriodInNanos) {
+
+            final Optional<RntbdContext> rntbdContext = requestManager.rntbdContext();
+            final int pendingRequestCount = requestManager.pendingRequestCount();
+
+            logger.warn("{} health check failed due to nonresponding read: {lastChannelWrite: {}, lastChannelRead: {}, "
+                + "readDelay: {}, readDelayLimit: {}, rntbdContext: {}, pendingRequestCount: {}}", channel,
+                timestamps.lastChannelWriteNanoTime(), timestamps.lastChannelReadNanoTime(), readDelay,
+                this.readDelayLimitInNanos, rntbdContext, pendingRequestCount);
+
+            String msg = MessageFormat.format(
+                "{0} health check failed due to nonresponding read: (lastChannelWrite: {1}, lastChannelRead: {2}, "
+                    + "readDelay: {3}, readDelayLimit: {4}, rntbdContext: {5}, pendingRequestCount: {6})", channel,
+                timestamps.lastChannelWriteNanoTime(), timestamps.lastChannelReadNanoTime(), readDelay,
+                this.readDelayLimitInNanos, rntbdContext, pendingRequestCount
+            );
+
+            return promise.setSuccess(msg);
+        }
+
+        if (this.idleConnectionTimeoutInNanos > 0L) {
+            if (currentTime - timestamps.lastChannelReadNanoTime() > this.idleConnectionTimeoutInNanos) {
```

Comment:
no - because the max time for write after read - like between request sent and first byte of response received is 10 seconds - compared ot idle time that is small time period. Also this logic is the same as it was in place - just added the reasons - no reason to change the logic (at least until we have more diagnostics)